EmpowerPlant/EmpowerPlantViewController.swift

```diff
@@ -8,21 +8,149 @@
+
+            // MARK: - Properties
+            let id: Int
+            let title: String
+            let description: String
+            let descriptionfull: String
+            let img: String
+            let imgcropped: String
+            let price: Int
+            // reviews: [{id: 4, productid: 4, rating: 4, customerid: null, description: null, created: String},...]
+
+        }
+        
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            if let data = data {
+                // Can't use 'Product' here because it can't be Decoded, so testing with Product1 for now
+                if let products = try? JSONDecoder().decode([Product1].self, from: data) {
+                    // Prints the 4 products from server, successfully
+                    for product in products {
+                        print(product.title, product.description)
+                    }
+                } else {
+                    print("Invalid Response")
+                }
+            } else if let error = error {
+                print("HTTP Request Failed \(error)")
+            }
+        }
+        // print("Dev Note - I am printed before the products info above is printed")
+        task.resume()
+        
+        // Don't Reload Table, because we still have to conver the above JSON objects from the server into Swift objects
+//        do {
+//            DispatchQueue.main.async {
+//                self.tableView.reloadData()
+//            }
+//        }
+//        catch {
+//            //error
+//        }
+    }
+    
+    func getAllProductsFromDb() {
+        do {
+            self.products = try context.fetch(Product.fetchRequest())
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+        catch {
+            // error
+        }
+    }
+    
+    func createProduct(title: String) {
+        let newProduct = Product(context: context)
+        newProduct.title = title
+        newProduct.text = "thedescription"
+        do {
+            try context.save()
+            getAllProductsFromDb()
+        }
+        catch {
+            // error
+        }
+    }
+
+    func deleteProduct(product: Product) {
+        context.delete(product)
+        do {
+            try context.save()
+        }
+        catch {
+            
+        }
+    }
+
+    func updateProduct(product: Product, newTitle:  String) {
+        product.title = newTitle
+        do {
+            try context.save()
+        }
+        catch {
+            
+        }
     }
     
     private func configureNavigationItems() {
+        
+        // TODO - put goToCart back eventually for the below #selector
         self.navigationItem.rightBarButtonItem = UIBarButtonItem(
             title: "Cart",
             style: .plain,
             target: self,
-            action: #selector(goToCart)
+            action: #selector(addToDb)
```

Comment:
needed a way to test adding to the DB, so temporarily used the Cart icon for that. But next PR I'll put it back to real navigation to the Cart.